src/main/java/com/auroraapi/models/Audio.java

```
@@ -1,4 +1,80 @@
 package com.auroraapi.models;
 
-public class Audio {
-}
+import javax.sound.sampled.*;
+import java.io.*;
+
+class Audio {
+  static final int NUM_CHANNELS = 1;
+  static final int SAMPLE_SIZE = 16;
+  static final int RATE = 16000;
+  static final boolean SIGNED = true;
+  static final boolean BIG_ENDIAN = true;
+
+  AudioFileFormat.Type fileType = AudioFileFormat.Type.WAVE;
+
+  // the line from which audio data is captured
+  TargetDataLine line;
+
+  AudioFormat format = new AudioFormat(RATE, SAMPLE_SIZE, NUM_CHANNELS, SIGNED, BIG_ENDIAN);
+
+  ByteArrayOutputStream data = new ByteArrayOutputStream();
+
+  void start() {
+    try {
+      DataLine.Info info = new DataLine.Info(TargetDataLine.class, format);
+
+      // checks if system supports the data line
+      if (!AudioSystem.isLineSupported(info)) {
+        System.out.println("Line not supported");
+        System.exit(0);
```

I don't think we should exit (it'll be unexpected to whatever developer is using the SDK). We should throw an exception instead or return a boolean indicating success or something so the developer can handle the case when there's no audio hardware present

---

Good call. Half of this was copy/pasted from an example and I didn't really think about all of it.